lib/Listeners/FileChanged.php

```diff
@@ -51,7 +51,7 @@ class FileChanged extends ListenersCore implements IEventListener {
 	 * @param Event $event
 	 */
 	public function handle(Event $event): void {
-		if (!($event instanceof NodeWrittenEvent)) {
+		if (!$this->registerFullTextSearchServices() || !($event instanceof NodeWrittenEvent)) {
```

Comment:
I'd flip the conditions here because if the event does not match we don't need to register anything?
```suggestion
		if (!($event instanceof NodeWrittenEvent) || !$this->registerFullTextSearchServices()) {
```